planetiler-core/src/main/java/com/onthegomap/planetiler/Planetiler.java

```diff
@@ -633,7 +636,12 @@ public void run() throws Exception {
 
     featureGroup.prepare();
 
-    MbtilesWriter.writeOutput(featureGroup, output, mbtilesMetadata, config, stats);
+    try (Mbtiles db = Mbtiles.newWriteToFileDatabase(output, config.compactDb())) {
```

Comment:
This is fine as-is since there's only one available output, but when we have more what do you think about ways to choose between the two? It could just go by file extension ".pmtiles" vs. ".mbtiles" but ultimately I think we'll want composable layers for input/output file types, encodings, and locations, for example:

- download from <s3/http/etc..>
- unwrap <zip/tar/tgz/bz2>
- read <shp/geopackage/flatgeobug/geojson/osm>
- then do planetiler stuff...
- output to <mbtiles/pmtiles>
- wrap <zip/tar/tgz/bz2>
- upload to <s3/http/...>

and each layer might have some parameters like s3 credentials/headers, mbtiles settings, gzip level, etc... I was thinking we could encode these as strings like `output.mbtiles?compact=false` or `input.shp.zip?src_proj=EPSG:3875` but not sure if you have any better ideas there?

Reply:
I don't think string encoding is going to work for specifying S3 endpoints, because those are fully formed URLs and will need to have a level of escaping to fit in string query parameters like that. Most `TileArchive` formats are going to be compressed internally so adding another wrapper of compression won't be that effective (any counterexamples, other than a dirtiles archive format?)

I say we just switch based on the output extension for now

Reply:
yeah switch on extension makes sense to start with.